### PR TITLE
SHOW SHARD GROUPS

### LINKED ISSUE #4728: SHOW SHARD GROUPS
To complement `SHOW SHARDS`, `SHOW SHARD GROUPS` would operate at the higher level, allowing groups of shards that represent the same time to be deleted.

### PR DESCRIPTION
**Pull request by mateuszdyminski**
This commit fixes #4728

### CODE REVIEW

**Comment by otoolep** [comment]
Thanks @mateuszdyminski -- have you signed the CLA?

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,7 @@
 
 ### Features
 - [#4790](https://github.com/influxdb/influxdb/pull/4790): Allow openTSDB point-level error logging to be disabled
+- [#4728](https://github.com/influxdb/influxdb/pull/4728): SHOW SHARD GROUPS
```
Comment (otoolep) [inline CHANGELOG.md]: Please add 'by @mateuszdyminski' to the end of the line. We like to include credits in the CHANGELOG.

File: influxql/ast.go [hunk 2]
```diff
@@ -2040,6 +2042,17 @@ func (s *ShowStatsStatement) RequiredPrivileges() ExecutionPrivileges {
 	return ExecutionPrivileges{{Admin: true, Name: "", Privilege: AllPrivileges}}
 }
 
+// ShowShardGroupsStatement represents a command for displaying groups of shard in the cluster.
```
Comment (otoolep) [inline influxql/ast.go]: Just say "shard groups" instead of "groups of shard". "shard groups" has a specific meaning.

File: influxql/ast.go [hunk 3]
```diff
@@ -2040,6 +2042,17 @@ func (s *ShowStatsStatement) RequiredPrivileges() ExecutionPrivileges {
...
+type ShowShardGroupsStatement struct{}
+
+// String returns a string representation.
```
Comment (otoolep) [inline influxql/ast.go]: "....of the SHOW SHARD GROUPS command".

File: influxql/parser_test.go [hunk 4]
```diff
@@ -1591,7 +1597,7 @@ func TestParser_ParseStatement(t *testing.T) {
 		{s: `SHOW RETENTION POLICIES`, err: `found EOF, expected ON at line 1, char 25`},
 		{s: `SHOW RETENTION POLICIES mydb`, err: `found mydb, expected ON at line 1, char 25`},
 		{s: `SHOW RETENTION POLICIES ON`, err: `found EOF, expected identifier at line 1, char 28`},
-		{s: `SHOW FOO`, err: `found FOO, expected CONTINUOUS, DATABASES, DIAGNOSTICS, FIELD, GRANTS, MEASUREMENTS, RETENTION, SERIES, SERVERS, SHARDS, STATS, SUBSCRIPTIONS, TAG, USERS at line 1, char 6`},
+		{s: `SHOW FOO`, err: `found FOO, expected CONTINUOUS, DATABASES, DIAGNOSTICS, FIELD, GRANTS, MEASUREMENTS, RETENTION, SERIES, SERVERS, SHARD, SHARDS, STATS, SUBSCRIPTIONS, TAG, USERS at line 1, char 6`},
```
Comment (otoolep) [inline influxql/parser_test.go]: We need an test case added to this list for `SHOW SHARD`.

File: meta/statement_executor_test.go [hunk 5]
```diff
@@ -966,6 +966,62 @@ func TestStatementExecutor_ExecuteStatement_Unsupported(t *testing.T) {
 }
 
 // Ensure a SHOW SHARDS statement can be executed.
```
Comment (otoolep) [inline meta/statement_executor_test.go]: Comment is not correct.

**Comment by otoolep** [comment]
Looks good @mateuszdyminski -- some small changes required.

@dgnorton -- can I get a +1 assuming requested changes are made?

**Comment by mateuszdyminski** [comment]
@otoolep I signed the CLA. 

All comments fixed.

**Comment by otoolep** [comment]
CLA confirmed signed.

@dgnorton -- please review and merge if it looks good to you.

**Comment by otoolep** [comment]
+1

**Comment by corylanou** [comment]
+1

**Comment by otoolep** [comment]
Thanks @mateuszdyminski !

**Comment by mateuszdyminski** [comment]
Thanks for help @otoolep. It was my first open-source contribution!

**Pull request merged**